base/unicode/utf8proc.jl

```diff
@@ -80,106 +39,60 @@ end
-    compat && (flags = flags | UTF8PROC_COMPAT)
+    stable && (flags = flags | STABLE)
+    compat && (flags = flags | COMPAT)
     if decompose
-        flags = flags | UTF8PROC_DECOMPOSE
+        flags = flags | DECOMPOSE
     elseif compose
-        flags = flags | UTF8PROC_COMPOSE
+        flags = flags | COMPOSE
     elseif compat || stripmark
         throw(ArgumentError("compat=true or stripmark=true require compose=true or decompose=true"))
     end
-    stripignore && (flags = flags | UTF8PROC_IGNORE)
-    rejectna && (flags = flags | UTF8PROC_REJECTNA)
-    newline2ls + newline2ps + newline2lf > 1 && throw(ArgumentError("only one newline conversion may be specified"))
-    newline2ls && (flags = flags | UTF8PROC_NLF2LS)
-    newline2ps && (flags = flags | UTF8PROC_NLF2PS)
-    newline2lf && (flags = flags | UTF8PROC_NLF2LF)
-    stripcc && (flags = flags | UTF8PROC_STRIPCC)
-    casefold && (flags = flags | UTF8PROC_CASEFOLD)
-    lump && (flags = flags | UTF8PROC_LUMP)
-    stripmark && (flags = flags | UTF8PROC_STRIPMARK)
+    stripignore && (flags = flags | IGNORE)
+    rejectna    && (flags = flags | REJECTNA)
+    newline2ls + newline2ps + newline2lf > 1 &&
+	throw(ArgumentError("only one newline conversion may be specified"))
+    newline2ls  && (flags = flags | NLF2LS)
+    newline2ps  && (flags = flags | NLF2PS)
+    newline2lf  && (flags = flags | NLF2LF)
+    stripcc     && (flags = flags | STRIPCC)
+    casefold    && (flags = flags | CASEFOLD)
+    lump        && (flags = flags | LUMP)
+    stripmark   && (flags = flags | STRIPMARK)
     utf8proc_map(s, flags)
 end
 
-function normalize_string(s::AbstractString, nf::Symbol)
-    utf8proc_map(s, nf == :NFC ? (UTF8PROC_STABLE | UTF8PROC_COMPOSE) :
-                    nf == :NFD ? (UTF8PROC_STABLE | UTF8PROC_DECOMPOSE) :
-                    nf == :NFKC ? (UTF8PROC_STABLE | UTF8PROC_COMPOSE
-                                   | UTF8PROC_COMPAT) :
-                    nf == :NFKD ? (UTF8PROC_STABLE | UTF8PROC_DECOMPOSE
-                                   | UTF8PROC_COMPAT) :
+normalize_string(s::AbstractString, nf::Symbol) =
+    utf8proc_map(s, nf == :NFC ? (STABLE | COMPOSE) :
+                    nf == :NFD ? (STABLE | DECOMPOSE) :
+                    nf == :NFKC ? (STABLE | COMPOSE | COMPAT) :
+                    nf == :NFKD ? (STABLE | DECOMPOSE | COMPAT) :
                     throw(ArgumentError(":$nf is not one of :NFC, :NFD, :NFKC, :NFKD")))
-end
 
 ############################################################################
 
 charwidth(c::Char) = Int(ccall(:utf8proc_charwidth, Cint, (UInt32,), c))
 
-lowercase(c::Char) = isascii(c) ? ('A' <= c <= 'Z' ? c + 0x20 : c) : Char(ccall(:utf8proc_tolower, UInt32, (UInt32,), c))
-uppercase(c::Char) = isascii(c) ? ('a' <= c <= 'z' ? c - 0x20 : c) : Char(ccall(:utf8proc_toupper, UInt32, (UInt32,), c))
+lowercase(c::Char) = (isascii(c)
+                      ? ('A' <= c <= 'Z' ? c + 0x20 : c)
```

Comment:
I had just added whitespace, because the old code was too long (I run into a lot of that, but I like keeping to 80-90 max per line). Since it lowers to the same code, I'll definitely start doing it that way, thanks for the info!